airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/manifest_component_transformer.py

```diff
@@ -89,60 +88,47 @@ def propagate_types_and_parameters(
         :param parent_parameters: The parameters set on parent components defined before the current component
         :return: A deep copy of the transformed component with types and parameters persisted to it
         """
-        propagated_component = dict(copy.deepcopy(declarative_component))
+        propagated_component = declarative_component.copy()
+
         if "type" not in propagated_component:
-            # If the component has class_name we assume that this is a reference to a custom component. This is a slight change to
-            # existing behavior because we originally allowed for either class or type to be specified. After the pydantic migration,
-            # class_name will only be a valid field on custom components and this change reflects that. I checked, and we currently
-            # have no low-code connectors that use class_name except for custom components.
             if "class_name" in propagated_component:
                 found_type = CUSTOM_COMPONENTS_MAPPING.get(parent_field_identifier)
             else:
                 found_type = DEFAULT_MODEL_TYPES.get(parent_field_identifier)
             if found_type:
                 propagated_component["type"] = found_type
 
-        # When there is no resolved type, we're not processing a component (likely a regular object) and don't need to propagate parameters
-        # When the type refers to a json schema, we're not processing a component as well. This check is currently imperfect as there could
-        # be json_schema are not objects but we believe this is not likely in our case because:
-        # * records are Mapping so objects hence SchemaLoader root should be an object
-        # * connection_specification is a Mapping
         if "type" not in propagated_component or self._is_json_schema_object(propagated_component):
```

Comment:
```suggestion
        # When there is no resolved type, we're not processing a component (likely a regular object) and don't need to propagate parameters
        # When the type refers to a json schema, we're not processing a component as well. This check is currently imperfect as there could
        # be json_schema are not objects but we believe this is not likely in our case because:
        # * records are Mapping so objects hence SchemaLoader root should be an object
        # * connection_specification is a Mapping
        if "type" not in propagated_component or self._is_json_schema_object(propagated_component):
```